src/main/java/com/ecommerce/galleon/controller/CartController.java

```diff
@@ -0,0 +1,78 @@
+package com.ecommerce.galleon.controller;
+
+
+import com.ecommerce.galleon.common.ApiResponse;
+import com.ecommerce.galleon.dto.cart.AddToCartDto;
+import com.ecommerce.galleon.dto.cart.CartDto;
+import com.ecommerce.galleon.model.User;
+import com.ecommerce.galleon.service.AuthenticationService;
+import com.ecommerce.galleon.service.CartService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/cart")
+public class CartController {
+
+    @Autowired
+    private CartService cartService;
+
+    @Autowired
+    private AuthenticationService authenticationService;
+
+
+    // post cart api
+    @PostMapping("/add")
+    public ResponseEntity<ApiResponse> addToCart(@RequestBody AddToCartDto addToCartDto,
+                                                 @RequestParam("token") String token) {
+        // authenticate the token
+        authenticationService.authenticate(token);
+
+
+        // find the user
+
+        User user = authenticationService.getUser(token);
+
+
+        cartService.addToCart(addToCartDto, user);
+
+        return new ResponseEntity<>(new ApiResponse(true, "Added to cart"), HttpStatus.CREATED);
+    }
+
+
+    // get all cart items for a user
+    @GetMapping("/")
+    public ResponseEntity<CartDto> getCartItems(@RequestParam("token") String token) {
+        // authenticate the token
+        authenticationService.authenticate(token);
+
+        // find the user
+        User user = authenticationService.getUser(token);
+
+        // get cart items
+
+        CartDto cartDto = cartService.listCartItems(user);
+        return new ResponseEntity<>(cartDto, HttpStatus.OK);
```

Comment:
Return **HttpStatus.OK** is not recommend because may **cartService.listCartItems(user)** returns an error or exception. did you handle this situation?